src/benchmark.cc

```diff
@@ -193,22 +203,36 @@ void RunInThread(const benchmark::internal::Benchmark::Instance* b,
   manager->NotifyThreadComplete();
 }
 
-std::vector<BenchmarkReporter::Run> RunBenchmark(
+struct RunResults {
+  std::vector<BenchmarkReporter::Run> non_aggregates;
+  std::vector<BenchmarkReporter::Run> aggregates_only;
+
+  bool display_report_aggregates_only;
+  bool file_report_aggregates_only;
+};
+
+RunResults RunBenchmark(
     const benchmark::internal::Benchmark::Instance& b,
     std::vector<BenchmarkReporter::Run>* complexity_reports) {
-  std::vector<BenchmarkReporter::Run> reports;  // return value
+  RunResults run_results;
 
   const bool has_explicit_iteration_count = b.iterations != 0;
   size_t iters = has_explicit_iteration_count ? b.iterations : 1;
   std::unique_ptr<internal::ThreadManager> manager;
   std::vector<std::thread> pool(b.threads - 1);
   const int repeats =
       b.repetitions != 0 ? b.repetitions : FLAGS_benchmark_repetitions;
-  const bool report_aggregates_only =
-      repeats != 1 &&
-      (b.aggregation_report_mode == internal::ARM_Unspecified
-           ? FLAGS_benchmark_report_aggregates_only
-           : b.aggregation_report_mode == internal::ARM_ReportAggregatesOnly);
+  run_results.display_report_aggregates_only =
+      repeats != 1 && (b.aggregation_report_mode == internal::ARM_Unspecified
+                           ? FLAGS_benchmark_report_aggregates_only ||
+                                 FLAGS_benchmark_display_aggregates_only
+                           : b.aggregation_report_mode &
+                                 internal::ARM_DisplayReportAggregatesOnly);
+  run_results.file_report_aggregates_only =
+      repeats != 1 && (b.aggregation_report_mode == internal::ARM_Unspecified
+                           ? FLAGS_benchmark_report_aggregates_only
+                           : b.aggregation_report_mode &
+                                 internal::ARM_FileReportAggregatesOnly);
```

Comment:
```c++
if (repeats != 1) {
  run_results.display_report_aggregates_only = (
      FLAGS_benchmark_report_aggregates_only || FLAGS_benchmark_display_aggregates_only);
  run_results.file_report_aggregates_only = FLAGS_benchmark_report_aggregates_only;
  if (b.aggregation_report_mode != internal::ARM_Unspecified) {
    run_results.display_report_aggregates_only = (
      b.aggregation_report_mode & internal::ARM_DisplayReportAggregatesOnly);
    run_results.file_report_aggregates_only = (
      b.aggregation_report_mode & internal::ARM_FileReportAggregatesOnly);
  }
}
```

a bit clearer than nested ternary operators, i think...

Reply:
Oh duh, I've meant to clean this but overlooked :)
Thanks!